ztp/tools/pgt2acm/packages/patches/patches.go

```diff
@@ -0,0 +1,292 @@
+		}
+	}
+
+	return nil
+}
+
+type KustomizeJSON struct {
+	Openapi   `yaml:"openapi"`
+	Patches   []Patch  `yaml:"patches"`
+	Resources []string `yaml:"resources"`
+}
+type Openapi struct {
+	Path string `yaml:"path"`
+}
+
+type Patch struct {
+	Path string `yaml:"path"`
+}
+
+type Resources []string
+
+// ApplyPatches applies the Kustomize patches on the input manifests using Kustomize and returns
+// the patched manifests. An error is returned if the patches can't be applied. This should be
+// run after the Validate method.
+func (m *ManifestPatcher) ApplyPatches(schema string) ([]map[string]interface{}, error) {
+	// Create the file system in memory with the Kustomize YAML files
+	fSys := filesys.MakeFsInMemory()
+
+	err := InitializeInMemoryKustomizeDir(fSys, schema)
+	if err != nil {
+		return nil, fmt.Errorf("failed to initialize Kustomize dir, err: %s", err)
+	}
+	kustomizationYAMLFile := KustomizeJSON{}
+	kustomizationYAMLFile.Openapi.Path = localSchemaFileName
+
+	options := []struct {
+		optionType   string
+		kustomizeKey string
+		objects      []map[string]interface{}
+	}{
+		{"manifest", "resources", m.Manifests},
+		{"patch", "patches", m.Patches},
+	}
+	for _, option := range options {
+		for i, object := range option.objects {
+			var objectYAML []byte
+			objectYAML, err := yaml.Marshal(object)
+			const errTemplate = "an unexpected error occurred when converting the %s back to " +
+				"YAML: %w"
+
+			if err != nil {
+				return nil, fmt.Errorf(errTemplate, option.optionType, err)
+			}
+
+			manifestFileName := fmt.Sprintf("%s%d.yaml", option.optionType, i)
+
+			err = fSys.WriteFile(path.Join(kustomizeDir, manifestFileName), objectYAML)
+			if err != nil {
+				return nil, fmt.Errorf(errTemplate, option.optionType, err)
+			}
+
+			if option.kustomizeKey == "patches" {
+				kustomizationYAMLFile.Patches = append(
+					kustomizationYAMLFile.Patches, Patch{Path: manifestFileName})
+			}
+			if option.kustomizeKey == "resources" {
+				kustomizationYAMLFile.Resources = append(
+					kustomizationYAMLFile.Resources, manifestFileName)
+			}
+		}
+	}
+
+	return KustomizeManifest(fSys, &kustomizationYAMLFile)
+}
+
+const (
+	localSchemaFileName = "schema.json"
+	kustomizeDir        = "kustomize"
+)
+
+// Initializes the in-memory file system with base directory and open API schema
+func InitializeInMemoryKustomizeDir(fSys filesys.FileSystem, schema string) (err error) {
+	err = fSys.Mkdir(kustomizeDir)
+	if err != nil {
+		return fmt.Errorf("an unexpected error occurred when configuring Kustomize: %w", err)
+	}
+	schemaJSON, err := os.ReadFile(schema)
+	if err != nil {
+		return fmt.Errorf("unable to open file: %s, err: %s ", schema, err)
+	}
+
+	err = fSys.WriteFile(path.Join(kustomizeDir, localSchemaFileName), schemaJSON)
+	if err != nil {
+		return fmt.Errorf("error writing schema, err:%s", err)
+	}
+	return nil
+}
+
+// Runs Kustomize to apply patches in the in-memory file system
```

Comment:
follow the func name description convention

Reply:
Fixed

Reply:
I don't see it was fixed, what i meant was to start with the description with the function name